core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -3913,17 +3920,36 @@ private RecordCursorFactory generateTableQuery0(
                         }
                     }
                 }
+                boolean orderByTimestamp = false;
+                // we can use skip sorting by timestamp if we:
+                // - query index with a single value or
+                // - query index with multiple values but use table order with forward scan (heap row cursor factory doesn't support backward scan) 
+                // it doesn't matter if we hit one or more partitions 
+                if (!orderByKeyColumn
+                        && isOrderByDesignatedTimestampOnly(model)
+                ) {
+                    int orderByDirection = getOrderByDirectionOrDefault(model, 0);
+                    if (nKeyValues == 1 || (nKeyValues > 1 && orderByDirection == ORDER_DIRECTION_ASCENDING)) {
+                        orderByTimestamp = true;
```

Comment:
Can this line 
```
orderByTimestamp = true;
```

go outside of this nested `if` since `isOrderByDesignatedTimestampOnly(model) == true` here?